proxy/src/coco/source.rs

```diff
@@ -0,0 +1,427 @@
+use serde::{Deserialize, Serialize};
+use std::collections::hash_map::DefaultHasher;
+use std::fmt;
+use std::hash::{Hash, Hasher};
+use std::str::FromStr;
+
+use librad::surf;
+use librad::surf::git::git2;
+
+use crate::error;
+
+use super::Peer;
+
+/// Branch name representation.
+#[derive(Clone, Debug, Eq, Ord, PartialEq, PartialOrd, Deserialize, Serialize)]
+pub struct Branch(pub(super) String);
```

Comment:
Some food for thought: I wonder what the advantage of `proxy` defining these types is. Would it be better to reuse the `surf` types and add de/serialization to them?